docs: using helm to generate k8s config

kaitmore opened the pull request:
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

<br/>
Resolves #353 
<br/>

2. What **changes to custom.yaml** are required?

<br/>n/a<br/>

3. Have you documented any additional setup steps or configurations options?

<br/>n/a<br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

<br/>n/a<br/>

chrisbsmith commented on docs/Generating k8s Config.md:
```diff
@@ -0,0 +1,42 @@
+# Generating k8s Configuration with Helm
+
+It is sometimes necessary to deploy Grey Matter without the use of Helm/Tiller. Luckily, Helm provides commands that can be used to generate raw kubernetes configuration.
```
I would say `without the use of Tiller`.  Helm is still used in this equation.

chrisbsmith commented on docs/Generating k8s Config.md:
````diff
@@ -0,0 +1,42 @@
+The template command can also be run on charts hosted in remote repositories, albeit not directly. To get around that, first fetch and untar the remote charts before running the templating command:
+
+```sh
+helm fetch --untar --untardir . decipher/greymatter
````
Do we need to add a line about this about adding the decipher helm repo?  Are these instructions expected to be a complete guide or a supplement to go along with the others?

kaitmore replied:
I linked to other documentation 👍

kaitmore commented:
@chrisbsmith Thanks for the comments! I've changed the file name and added more prereqs/links to the minikube doc. I figure this doc will be more of a supplement.

kaitmore commented:
@tilleryd 

`helm template` runs through all charts and templates out every file. After looking through helm docs I don't see any way to tell it not to template out specific files, tests or otherwise. 

I could either put a note in the doc saying to delete connection tests if not desired, or I could add a variable to each test yaml, such as `{{- if .gm-control-api.enableTest }}`, although that might be out of scope for this PR.

tilleryd commented:
I think we should add a variable to turn off the tests. They haven't been a factor since we've been deploying the charts. @chrisbsmith I'm curious on your thoughts here.

kaitmore commented:
@tilleryd My only other hesitation is that "turning off tests" is a bit misleading. Running `helm install` doesn't run tests - so setting the `enableTests` value wouldn't do anything in that scenario, it would only be used for this specific case where we're templating.

chrisbsmith commented:
I will say that we took an initial dive into using the Helm tests as an opportunity to see if it would be useful for us.  It may be in the long term, but we have not done anything to keep these files up to date.  You can see that it's trying to find catalog at `/services/catalog/1.2`.  I'd be OK removing the files all together but possibly doing more research into leveraging the `helm test` feature.  

@danielpcox  may be interested in this as well

tilleryd commented:
I say we delete them since gm-integrity will satisfy the mesh verification tests.

chrisbsmith commented:
no issues here, but I do think the test hook may be a useful feature to deploy gm-integrity into k8s automatically.
